WalletWasabi.Fluent/ViewModels/SearchBar/Sources/TransactionsSearchSource.cs

```diff
@@ -117,6 +120,28 @@ private static IEnumerable<ISearchItem> Search(string query)
 	private static IEnumerable<(WalletViewModel, HistoryItemViewModelBase)> Filter(string queryStr)
 	{
 		return Flatten(GetTransactionsByWallet())
-			.Where(tuple => ContainsId(tuple.Item2, queryStr));
+		.Where(tuple => TryParseBitcoinAddress(tuple.Item1.WalletModel.Network, queryStr, out var address) ?
+			ContainsDestinationAddress(tuple.Item1, tuple.Item2, address) :
+			ContainsId(tuple.Item2, queryStr));
+	}
+
+	private static bool ContainsDestinationAddress(WalletViewModel walletViewModel, HistoryItemViewModelBase historyItem, BitcoinAddress address)
+	{
+		var txid = historyItem.Transaction.Id;
+		return walletViewModel.WalletModel.Transactions.GetDestinationAddresses(txid).Contains(address);
+	}
+
+	private static bool TryParseBitcoinAddress(Network network, string queryStr, [NotNullWhen(true)] out BitcoinAddress? address)
+	{
+		address = null;
+		try
+		{
+			address = BitcoinAddress.Create(queryStr, network);
+			return true;
+		}
+		catch (FormatException)
+		{
+			return false;
+		}
```

Comment:
@lontivero's suggestion was to move this method to `NBitcoinHelpers` and also rename it. https://github.com/zkSNACKs/WalletWasabi/pull/12604#discussion_r1514415369